service/rest_task_history.go

```diff
@@ -168,9 +168,13 @@ func (restapi restAPI) GetTestHistory(w http.ResponseWriter, r *http.Request) {
 			if result.TaskTimedOut {
 				taskStatus = model.TaskTimeout
 			}
-			if result.TaskDetailsType == "system" {
+			// TODO should this elseif?
+			if result.TaskDetailsType == model.SystemCommandType {
 				taskStatus = model.TaskSystemFailure
 			}
+			if result.TaskDetailsType == model.SetupCommandType {
+				taskStatus = model.TaskSetupFailure
+			}
```

Comment:
Since `result.TaskDetailsType` couldn't be equal to both `model.*` values, it might be replaced with elseif. I still does now know all the go syntax keywords, but if it has `switch` or pattern matching it might be the best option, per my vision.

Reply:
Oh that comment was in a bad spot, it was actually referring to just above it. As written, taskStatus is always overwritten to Setup or System failure if it timed out, but the type of the task was system or setup.